src/components/input/input.ts

```diff
@@ -0,0 +1,35 @@
+import "./input.css";
+import template from "./input.hbs";
+import { Block } from "../../core/Block";
+
+interface Props {
+  name?: string;
+  type?: string;
```

Comment:
тут лучше использовать union
```suggestion
  type: "number" | "email" и тд;
```

Reply:
предполагаю возможным "string" | "number" | "email" | "tel" | "password"